runtime/vm/jfr.cpp

```diff
@@ -1152,6 +1160,134 @@ jfrDump(J9VMThread *currentThread, BOOLEAN finalWrite)
 	flushAllThreadBuffers(currentThread, finalWrite);
 	writeOutGlobalBuffer(currentThread, finalWrite);
 }
+
+UDATA
+jfrTypeIDHashFn(void *key, void *userData)
+{
+	J9JFRTypeID *entry = (J9JFRTypeID *) key;
+
+	U_16 length = J9UTF8_LENGTH(entry->className);
+	U_8 *data = J9UTF8_DATA(entry->className);
+
+	/* sdbm algorithm */
+	UDATA hash = 0;
+
+	for (U_16 i = 0; i < length; i++) {
+		hash = data[i] + (hash << 6) + (hash << 16) - hash;
+	}
+
+	return hash;
+}
+
+UDATA
+jfrTypeIDHashEqualFn(void *tableNode, void *queryNode, void *userData)
+{
+	J9JFRTypeID *tableEntry = (J9JFRTypeID *) tableNode;
+	J9JFRTypeID *queryEntry = (J9JFRTypeID *) queryNode;
+
+	return J9UTF8_EQUALS(tableEntry->className, queryEntry->className);
+}
+
+UDATA
+initializeJFRIDs(J9JavaVM *vm)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	UDATA result = 0;
+
+	vm->jfrState.typeIDs = hashTableNew(OMRPORT_FROM_J9PORT(privatePortLibrary), J9_GET_CALLSITE(), 0, sizeof(J9JFRTypeID), sizeof(J9JFRTypeID *), 0, J9MEM_CATEGORY_CLASSES, jfrTypeIDHashFn, jfrTypeIDHashEqualFn, NULL, NULL);
+	if (NULL == vm->jfrState.typeIDs) {
+		result = 1;
+		goto done;
+	}
+
+	if (omrthread_monitor_init_with_name(&vm->jfrState.typeIDMonitor, 0, "JFR Type ID monitor")) {
+		result = 1;
+		goto done;
+	}
+
+	/* Value needs to be the same as jdk.jfr.internal.JVM.RESERVED_CLASS_ID_LIMIT */
+	vm->jfrState.typeIDcount = 500;
+
+done:
+	return result;
+}
+
+UDATA
+freeJFRIDStrings(void *entry, void *userData)
+{
+	J9JavaVM *vm = (J9JavaVM*)userData;
+	J9JFRTypeID *jfrTypeID = (J9JFRTypeID*)entry;
```

Comment:
Please format casts consistently. Some places there's a space after `)`, but not here.
Ideally, there'd also be a space before `*`.